From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 19 Sep 2016 18:07:24 +0100
Subject: Revert "s390: Define AT_VECTOR_SIZE_ARCH for ARCH_DLINFO"
Forwarded: not-needed

This reverts commit 9693336d43ac21155438cd084f02cfaa94ecaddf which
was commit 68c5cf5a6091c2c3fabccfd42ca844d730ec24c6 upstream.
The bug doesn't cause a real problem yet and the fix changes
struct mm_struct which breaks the module ABI.

---
--- a/arch/s390/include/asm/elf.h
+++ b/arch/s390/include/asm/elf.h
@@ -210,7 +210,6 @@ do {								\
 
 #define STACK_RND_MASK	0x7ffUL
 
-/* update AT_VECTOR_SIZE_ARCH if the number of NEW_AUX_ENT entries changes */
 #define ARCH_DLINFO							    \
 do {									    \
 	if (vdso_enabled)						    \
--- a/arch/s390/include/uapi/asm/auxvec.h
+++ b/arch/s390/include/uapi/asm/auxvec.h
@@ -3,6 +3,4 @@
 
 #define AT_SYSINFO_EHDR		33
 
-#define AT_VECTOR_SIZE_ARCH 1 /* entries in ARCH_DLINFO */
-
 #endif
